onnxruntime/core/session/inference_session.cc

```diff
@@ -546,6 +546,48 @@ common::Status InferenceSession::InitializeSubgraphSessions(Graph& graph, Sessio
   return Status::OK();
 }
 
+static void ModelUseFP16Helper(const onnx::TypeProto type_proto, bool& use_fp16) {
```

Comment:
Code could be clearer if `use_fp16` was returned by value, and there is no performance benefit to using a reference argument since it's a small, trivial type.